app/test_main.py

```diff
@@ -2,0 +1,39 @@
+import pytest
+from unittest import mock
+from unittest.mock import MagicMock
+
+
+from app.main import can_access_google_page
+
+
+@pytest.mark.parametrize(
+    "url,mocked_result_connection,mocked_result_url,result",
+    [
+        ("http://google.com", True, True, "Accessible"),
+        ("http://google.com", False, False, "Not accessible"),
+        ("http://google.com", True, False, "Not accessible"),
+        ("http://google.com", False, True, "Not accessible"),
+    ],
+    ids=[
+        "return 'Accessible' connection is working, url is valid",
+        "return 'Not accessible' connection is not working, url do not valid",
+        "return 'Not accessible' connection is working, url do not valid",
+        "return 'Not accessible' connection is not working, url is valid"
+    ]
+)
+@mock.patch("app.main.valid_google_url")
+@mock.patch("app.main.has_internet_connection")
+def test_has_internet_connection_exist(
```

Comment:
The test function name 'test_has_internet_connection_exist' is not descriptive enough. Consider renaming it to something like 'test_can_access_google_page' to better reflect what the test is verifying.